File: Algorithms/541.Reverse-String-II/reverse_string_ii.go
func reverseStr(s string, k int) string {
    byte_s := []byte(s)
    length := len(byte_s)
    for i:=0; i<length; i+=2*k {
        x, y := i, i+k-1
        if y > length-1 {
            y = length - 1
        }
        for ; x<y; x, y = x+1, y-1 {
            byte_s[x], byte_s[y] = byte_s[y], byte_s[x]
        }
    }
    return string(byte_s)
}

File: Algorithms/541.Reverse-String-II/reverse_string_ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        p = 0
        length = len(s)
        while p*k < length:
            s = s[:p*k] + s[p*k: p*k+k][::-1] + s[p*k+k:]
            p += 2
        return s
